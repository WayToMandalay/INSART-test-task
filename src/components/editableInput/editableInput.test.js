import React, { useState } from 'react'
import { render, screen, cleanup, fireEvent } from '@testing-library/react'
import renderer from 'react-test-renderer'
import '@testing-library/jest-dom'
import EditableInput from './editableInput'

afterEach(() => {
    cleanup()
})

const params = {
    type: 'sale',
    label: 'Sale',
    basicValue: 40,
}

render(<EditableInput {...params} />)
const inputBlock = screen.getByTestId('input-1')
const inputEl = screen.getByTestId('input-2')

describe('Input Test Block', () => {
    test('Base Tests', () => {
        expect(inputBlock).toBeInTheDocument()
        expect(inputEl).toBeInTheDocument()
        expect(inputBlock).toHaveTextContent('Sale')
    })
    test('Base Tests 2', () => {
        fireEvent.change(inputEl, { target: { value: 390 } })

        expect(inputEl.value).toBe('390')
        console.log(inputEl)
        expect(inputEl).toHaveClass('isInvalid')
    })
})

test('matches snapshot', () => {
    const params = {
        type: 'sale',
        label: 'Sale',
        basicValue: 39.2,
    }
    const tree = renderer.create(<EditableInput {...params} />).toJSON()
    expect(tree).toMatchSnapshot()
})
