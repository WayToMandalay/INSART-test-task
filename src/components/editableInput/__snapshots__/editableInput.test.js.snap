// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="editableInput__container"
  data-testid="input-1"
>
  <label>
    Sale
    <input
      className="editableInput"
      data-testid="input-2"
      defaultValue={39.2}
      onChange={[Function]}
      onFocus={[Function]}
    />
    <span />
  </label>
</div>
`;
